app/src/test/java/ch/epfl/sweng/eventmanager/ui/eventShowcase/models/ScheduleViewModelTest.java

```diff
@@ -0,0 +1,107 @@
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+import org.mockito.junit.MockitoJUnitRunner;
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.UUID;
+
+import static org.junit.Assert.*;
+import static org.mockito.ArgumentMatchers.anyInt;
+import static org.mockito.Mockito.mock;
+
+@RunWith(MockitoJUnitRunner.class)
+public class ScheduleViewModelTest {
+
+    @Rule
+    public InstantTaskExecutorRule instantTaskExecutorRule = new InstantTaskExecutorRule();
+    @Mock
+    private EventRepository eventRepository;
+    @Mock
+    private JoinedScheduleItemRepository joinedScheduleItemRepository;
+
+    private ScheduleViewModel scheduleViewModel;
+    private ArrayList<ScheduledItem> concerts = new ArrayList<>();
+    private ArrayList<JoinedScheduleItem> joinedScheduleItems = new ArrayList<>();
+    private MutableLiveData<List<ScheduledItem>> data = new MutableLiveData<>();
+    private MutableLiveData<List<JoinedScheduleItem>> joinedData = new MutableLiveData<>();
+    private ArrayList<ScheduledItem> polyv = new ArrayList<>();
+
+    @Before
+    public void setUp() {
+        initMock();
+        MockitoAnnotations.initMocks(this);
+        Mockito.when(eventRepository.getScheduledItems(anyInt())).thenReturn(data);
+        Mockito.when(joinedScheduleItemRepository.findByEventId(anyInt())).thenReturn(joinedData);
+        scheduleViewModel = new ScheduleViewModel(eventRepository,joinedScheduleItemRepository);
+        scheduleViewModel.init(anyInt());
+        Observer obs = mock(Observer.class);
+        scheduleViewModel.getJoinedScheduleItems().observeForever(obs);
+        scheduleViewModel.getScheduleItemsByRoom().observeForever(obs);
+    }
+
+    @Test
+    public void getScheduledItems() {
+        assertEquals(data.getValue(), scheduleViewModel.getScheduledItems().getValue());
+    }
+
+    @Test
+    public void getScheduleItemsByRoom() {
+        System.out.println(scheduleViewModel.getScheduleItemsByRoom().getValue());
+        assertEquals(polyv,scheduleViewModel.getScheduleItemsByRoom().getValue().get("Polyv"));
+    }
+
+    @Test
+    public void isItemJoined() {
+        assertTrue(scheduleViewModel.isItemJoined(joinedScheduleItems.get(0).getUid()));
+    }
+
+    @Test
+    public void getJoinedScheduleItems() {
+        assertEquals(joinedScheduleItems.get(0).getUid(),scheduleViewModel.getJoinedScheduleItems().getValue().get(0).getId());
+    }
+
+    @Test
+    public void toggleMySchedule() {
+        Context context = Mockito.mock(Context.class);
+        scheduleViewModel.toggleMySchedule(concerts.get(1).getId(),context);
+        assert(scheduleViewModel.getJoinedScheduleItems().getValue().contains(concerts.get(1)));
+    }
+
+
+    private void initMock() {
```

Comment:
Can't you use `MockConcertRepository` here? Copy-pasted code is going to bite us at some point.

Reply:
Yes you're right I could but MockConcertRepository was designed at first to be used in UI test and therefore is in AndroidTest package and was not in the scope of the test package when I wrote this test but I've copied the class and it's fixed now

Reply:
Your comment is rather hard to understand, but this code is still duplicated (present in both tests and androidTests!).

Reply:
Yes because I duplicated the code as you asked me to be able to use the mockConcertRepo in the test package but it was not there before :)

Reply:
Why can't you use the same class for both test and androidTest?

Reply:
Because if you try you'll see that the classes between test and androidTest can't be shared...

Reply:
Why? Please provide a reference to the documentation.

Reply:
I'm sorry I don't know why and I don't have link for documentation right now to give you but there are several posts on Stackoverflow where they talk about ways to fix this issue by creating package for shared class in src. However I didn't like this way of doing it so I preferred to just duplicate the class

Reply:
Are you talking about things like https://stackoverflow.com/questions/39519347/share-test-classes-between-test-and-androidtest#44138801 ? It looks ways cleaner to me, can you detail your "I don't like it" ?

Reply:
Yes I'm talking about that. Because it's a test class so it's more coherent for it to be in the respective test packages. Furthermore, when I wrote the class, it was in the src folder in a mock package and it was moved by other members in the test folder. At least to me it's not a big issue so we can always fix that later if that bothers you

Reply:
I don't aggree with you, copy-pasted code is a NO-GO to me. I need more to be convinced.

@zyuiop @antoine-crettenand can you give your opinion?

Reply:
I think you can put a TODO and resolve